extensions_admin/pulp_ostree/extensions/admin/cudl.py

```diff
@@ -49,15 +59,27 @@ def _describe_distributors(self, user_input):
         is needed to create distributor configs.
 
         :param user_input:  dictionary of data passed in by okaara
-        :type  user_inpus:  dict
+        :type  user_input:  dict
 
         :return:    list of dict containing distributor_type_id,
                     repo_plugin_config, auto_publish, and distributor_id (the same
                     that would be passed to the RepoDistributorAPI.create call).
         :rtype:     list of dict
         """
-        config = {}
-        auto_publish = user_input.get('auto-publish', True)
+        relative_path = user_input.get(OPT_RELATIVE_PATH.keyword)
+        auto_publish = user_input.get(OPT_AUTO_PUBLISH.keyword, True)
+
+        # relative path derived using the path component of the feel url when not specified
+        if not relative_path:
+            feed_url = user_input.get(self.options_bundle.opt_feed.keyword)
+            if feed_url:
+                url = urlparse(feed_url)
+                relative_path = url.path
```

Comment:
It seems like this logic should be built into the distributor instead of being here. What will happen if a user updates the feed_url, but doesn't update the relative path? It seems like their relative path might get overridden, which I would think would be a surprising behavior.

If we move the logic to be at publish time in the distributor, then it has two benefits:
0) It doesn't permanently set the relative path, but only derives it when it is not otherwise set
1) It will also work for REST API users.